library/alloc/src/gc.rs

```diff
@@ -124,6 +124,12 @@ impl<T: ?Sized> Gc<T> {
         this.ptr.as_ptr() as *const T
     }
 
+    /// Get a raw pointer to the underlying value `T`.
+    #[unstable(feature = "gc", issue = "none")]
+    pub fn as_ptr(this: Self) -> *const T {
+        this.ptr.as_ptr() as *const T
```

Comment:
Why do we do a cast here? Doesn't `as_ptr()` already give us a pointer type?

Reply:
Ahh, ignore me, I see it returns a `*mut T`. We don't want that!

Reply:
My former self had his reasons!

Reply:
Ha!